golink.go

```diff
@@ -169,18 +170,78 @@ func Run() error {
 	if err := srv.Start(); err != nil {
 		return err
 	}
-	localClient, _ = srv.LocalClient()
 
-	l80, err := srv.Listen("tcp", ":80")
+	// create tsNet server and wait for it to be ready & connected.
+	localClient, _ = srv.LocalClient()
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	_, err = srv.Up(ctx)
 	if err != nil {
 		return err
 	}
 
-	log.Printf("Serving http://%s/ ...", *hostname)
-	if err := http.Serve(l80, serveHandler()); err != nil {
-		return err
+	enableTLS := len(srv.CertDomains()) > 0
+	if enableTLS {
+		// warm the certificate cache for all cert domains to prevent users waiting
+		// on ACME challenges in-line on their first request.
+		for _, d := range srv.CertDomains() {
+			log.Printf("Provisioning TLS certificate for %s ...", d)
+			ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
```

Comment:
what happens when this times out? For example, if we're not able to complete the DNS challenge to provision the cert?